route.go

```diff
@@ -0,0 +1,67 @@
+package fuego
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+func NewRoute[T, B any](method, path string, handler any, openapi *OpenAPI, options ...func(*BaseRoute)) Route[T, B] {
+	return Route[T, B]{
+		BaseRoute: NewBaseRoute(method, path, handler, openapi, options...),
+	}
+}
+
+// Route is the main struct for a route in Fuego.
+// It contains the OpenAPI operation and other metadata.
+// It is a wrapper around BaseRoute, with the addition of the response and request body types.
+type Route[ResponseBody any, RequestBody any] struct {
+	BaseRoute
+}
+
+func NewBaseRoute(method, path string, handler any, openapi *OpenAPI, options ...func(*BaseRoute)) BaseRoute {
+	baseRoute := BaseRoute{
+		Method:    method,
+		Path:      path,
+		Params:    make(map[string]OpenAPIParam),
+		FullName:  FuncName(handler),
+		Operation: openapi3.NewOperation(),
+		OpenAPI:   openapi,
+	}
+
+	for _, o := range options {
+		o(&baseRoute)
+	}
+
+	return baseRoute
+}
+
+// BaseRoute is the base struct for all routes in Fuego.
+// It contains the OpenAPI operation and other metadata.
+type BaseRoute struct {
+	Operation            *openapi3.Operation // GENERATED OpenAPI operation, do not set manually in Register function. You can change it after the route is registered.
```

Comment:
Nitpick: I've always been a fan of the comment above the field

Reply:
Yeah it allows easier multiline read. I've just copy-pasted this one but I'll try to migrate all comments above the field in the next PRs

Reply:
Fixed. Finally I'm merging, as your PR does not collide with this one :)